core/interfaces/i_connection_previewer.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import type {BlockSvg} from '../block_svg';
+import type {RenderedConnection} from '../rendered_connection';
+
+/**
+ * Displays visual "previews" of where a block will be connected if it is
+ * dropped.
+ */
+export interface IConnectionPreviewer {
+  /**
+   * Display a connection preview where the draggedCon connects to the
+   * staticCon, replacing the replacedBlock (currently connected to the
+   * staticCon).
+   *
+   * @param draggedCon The connection on the block stack being dragged.
+   * @param staticCon The connection not being dragged that we are
+   *     connecting to.
+   * @param replacedBlock The block currently connected to the staticCon that
+   *     is being replaced.
+   */
+  previewReplacement(
+    draggedCon: RenderedConnection,
```

Comment:
nit (and you can disagree with me here) but I prefer "conn" as the abbreviation for "connection" if we're going to abbreviate it at all. apologies as i meant to leave this comment in your doc but hoping i've caught you early enough here

Reply:
Oh yeah totally good! I did the types in a separate PR partially so that it would be easier to bike shed :P So this is perf!